### DISCUSSION TITLE
local storage access

### DISCUSSION THREAD

**Discussion by pokemonbattledroid** [lrstanley/arlo-dl#8] [General]
Hi,

I came across this as I tried to find a solution to my frustration of not being able to access local storage recordings, unless I physically unplug the USB drive from my basestation (VMB4000).  Newer devices seem to have some feature but I refuse to upgrade a system when cameras are backwards compatible and the whole point of the system is to monitor and record - but seems to be missing the point for me.

Does this feature, arlo-dl, allow me to access the USB on the network?  SSH into the base station, or...?

**Comment by lrstanley** [comment]
arlo-dl simply interacts with the cloud API, to save all cloud videos to a local filesystem. This means that the only thing that is interacted with are cloud videos, not the actual appliance on the local network.

However, as long as arlo-dl is ran at least every 7 days, it will mean that all videos should be saved (even if those videos expire after 7 days or so in the cloud environment).

**Comment by pokemonbattledroid** [comment]
Is there anyway to use this in a way to access the local storage? It seems
like it is accessing it somehow to save the cloud files, if they are
already saved locally, might there be a way to see them from a local lan?

**Reply by lrstanley** [reply]
The way the cloud videos work, is the hub sends those to their cloud infrastructure, and arlo-dl simply downloads them from the cloud. Things like files on the local USB drive connected to the hub, don't get sent up and/or aren't accessible, and thus I don't think there would be a way for arlo-dl to see them.

See below diagram:

![](https://cdn.liam.sh/share/2022/11/firefox_i9N1Bi9dmy.png)

-----------

If there were some kind of API to achieve what you're asking, I'm not familiar with it. Arlo APIs are completely undocumented and the libraries that `arlo-dl` relies on are possible due to packet capture/inspection, so even if it was possible, it'd likely be quite a bit of effort to accomplish.

**Comment by pokemonbattledroid** [comment]
Right, quite complex for me to understand, but it sounds like it's not so
easy.  I wish there was just a way to access the USB drive as I don't have
a subscription whereby it's sent to the cloud.  A slight nuisance to get to
the hub each time to eject drive and then plug into computer to see the
video...  Hope there is some sort of resource out there...

Thanks!
--
iamgo.net

**Reply by lrstanley** [reply]
For what it's worth, I don't pay for a subscription (and just have the free 7 day cloud storage or whatever it is), and just run arlo-dl frequently enough that the videos get mirrored from the cloud (before they're removed) to my local system/NAS.

**Comment by pokemonbattledroid** [comment]
Thanks.  I do have that too, but when I purchased a new camera, it seems
that one isn't part of the free 7 day legacy plan :(

I'm now looking into these home automation things and maybe there is a clue
in there, but seems a bigger project.  Short of just subscribing, I think
there is something out there!
--
iamgo.net
